packages/upload-client/src/blob.js

```diff
@@ -26,6 +27,40 @@ function createUploadProgressHandler(url, handler) {
   return onUploadProgress
 }
 
+// FIXME this code was copied over from w3up-client and modified to parameterise receiptsEndpoint
```

Comment:
I though I wrote this earlier, but can't find it now. Can we make w3up to use this function from `upload-client` instead of having same code in both places?

Reply:
addressed. I think I somehow mixed it with doing the same for the receipts endpoint